nncf/tensorflow/quantization/algorithm.py

```diff
@@ -153,40 +267,53 @@ def get_transformation_layout(self, model: tf.keras.Model) -> TFTransformationLa
+                target_point = TFLayerWeight(target_layer_name, weight_def.weight_attr_name)
+                qpoint = QuantizationPoint(op_name, quantizer_spec, target_point)
+                setup.add_quantization_point(qpoint)
 
         insertion_points = self._find_insertion_points(nncf_graph)
         qconfig = self._get_default_qconfig(self.global_quantizer_constraints[QuantizerGroup.ACTIVATIONS])
         for original_node_name, instance_index in insertion_points:
             fake_quantize_name = self._get_fake_quantize_name(original_node_name, instance_index)
-            fake_quantize_layer = FakeQuantize(
-                TFQuantizerSpec.from_config(qconfig, narrow_range=False, half_range=False),
-                name=fake_quantize_name)
-            self._op_names.append(fake_quantize_layer.op_name)
-
-            transformations.register(
-                TFInsertionCommand(
-                    target_point=TFAfterLayer(original_node_name, instance_index),
-                    callable_object=fake_quantize_layer,
-                    priority=TransformationPriority.QUANTIZATION_PRIORITY))
+            quantizer_spec = TFQuantizerSpec.from_config(qconfig, narrow_range=False, half_range=False)
+            target_point = TFAfterLayer(original_node_name, instance_index)
+            qpoint = QuantizationPoint(fake_quantize_name, quantizer_spec, target_point)
+            setup.add_quantization_point(qpoint)
+        return setup
+
+    def _build_insertion_commands_for_quantizer_setup(self,
+                                                      quantizer_setup: QuantizationSetup) -> List[TFInsertionCommand]:
+        insertion_commands = []
+        for quantization_point in quantizer_setup:
+            op_name = quantization_point.op_name
+            quantizer_spec = quantization_point.quantizer_spec
+            target_point = quantization_point.target_point
+            if quantization_point.is_weight_quantization():
+                quantizer = self._create_quantizer(op_name, quantizer_spec)
+            else:
+                quantizer = FakeQuantize(quantizer_spec, name=op_name)
+                self._op_names.append(quantizer.op_name)
+            command = TFInsertionCommand(target_point=target_point,
+                                         callable_object=quantizer,
+                                         priority=TransformationPriority.QUANTIZATION_PRIORITY)
+            insertion_commands.append(command)
+        return insertion_commands
 
+    def get_transformation_layout(self, model: tf.keras.Model) -> TFTransformationLayout:
+        if self._quantizer_setup is None:
+            self._quantizer_setup = self._get_quantizer_setup(model)
+        transformations = TFTransformationLayout()
+        insertion_commands = self._build_insertion_commands_for_quantizer_setup(self._quantizer_setup)
+        for command in insertion_commands:
+            transformations.register(command)
```

Comment:
Refactored quantization builder to explicitly generate setup, build insertion commands and register them one-by-one similar to PT. @vshampor